src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Helpers.cs

```diff
@@ -97,6 +97,11 @@ private static void WriteValueCore(Utf8JsonWriter writer, object value, Type typ
                 options = JsonSerializerOptions.s_defaultOptions;
             }
 
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
```

Comment:
Should this be checked here, or in a public method elsewhere?

Reply:
Both public methods call into this method so was following the pattern in the existing code that defers some of the arg validation to the common helper (similar to options/null stream/etc.). Otherwise, I would have done it in the public method.

I left refactoring out of this PR so I can port this change to 3.1.